File: internal/content/testdata/bench/go/subset-sum-equals-target.go
func subset_sum_equals_target(nums []int, target int) bool {
	dp := make([]bool, target+1)
	dp[0] = true
	for _, num := range nums {
		for j := target; j >= num; j-- {
			dp[j] = dp[j] || dp[j-num]
		}
	}
	return dp[target]
}

File: internal/content/testdata/bench/rust/subset-sum-equals-target.rs
fn subset_sum_equals_target(nums: Vec<i64>, target: i64) -> bool {
    if target < 0 {
        return false;
    }
    let t = target as usize;
    let mut dp = vec![false; t + 1];
    dp[0] = true;
    for &num in &nums {
        if num < 0 {
            continue;
        }
        let num_u = num as usize;
        if num_u > t {
            continue;
        }
        for j in (num_u..=t).rev() {
            dp[j] = dp[j] || dp[j - num_u];
        }
    }
    dp[t]
}
